### PR TITLE
Use separate feature flags for admin and applicant logout

### PR DESCRIPTION
**Pull request by yotommy**
### Description

Replace `ENHANCED_OIDC_LOGOUT_ENABLED` feature flag with separate feature flags for admin vs applicant.

This is required if a deployment uses OIDC for both applicants and admins, but one identity provider requires the extra logout param and the other one prohibits it.

The PR also adds some plumbing to get the flags where they need to go. Subsequent PRs will add new logic that is gated by these flags.

## Release notes

This PR introduces two new feature flags:

- `ADMIN_OIDC_ENHANCED_LOGOUT_ENABLED`
- `APPLICANT_OIDC_ENHANCED_LOGOUT_ENABLED`

These will gate new functionality and should remain **disabled** until a future release.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

### Issue(s) this completes

Relates to #4359

### CODE REVIEW

**Comment by yotommy** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5832** <a href="https://app.graphite.dev/github/pr/civiform/civiform/5832" target="_blank">Graphite</a>  👈
    * **PR #5730** <a href="https://app.graphite.dev/github/pr/civiform/civiform/5730" target="_blank">Graphite</a> 
      * **PR #5731** <a href="https://app.graphite.dev/github/pr/civiform/civiform/5731" target="_blank">Graphite</a> 
        * **PR #5816** <a href="https://app.graphite.dev/github/pr/civiform/civiform/5816" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/5832?utm_source=stack-comment).

File: server/conf/env-var-docs.json [hunk 1]
```diff
@@ -713,9 +713,14 @@
         "description": "Enables caching for questions and their associated data.",
         "type": "bool"
       },
-      "ENHANCED_OIDC_LOGOUT_ENABLED": {
-        "mode": "HIDDEN",
-        "description": "Enables logic to populate more fields in OIDC logout requests.",
+      "ADMIN_OIDC_ENHANCED_LOGOUT_ENABLED": {
+        "mode": "ADMIN_WRITEABLE",
+        "description": "Enables logic to populate more fields in OIDC logout requests to admin identity provider.",
```
Comment (bion) [inline server/conf/env-var-docs.json:718]: **nitpick (non-blocking):** "logic to" is redundant
```suggestion
        "description": "Enables populating more fields in OIDC logout requests to admin identity provider.",
```

**Review by bion — APPROVED** [review]

File: server/conf/env-var-docs.json (hunk 1 above)
Comment (yotommy) [reply server/conf/env-var-docs.json:718]: Done

**Comment by yotommy** [comment]
### Merge activity

* **Nov 14, 9:20 PM**: @yotommy merged this pull request with [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/5832).

**Pull request merged**